### PR TITLE
Avoid argocd login

### LINKED ISSUE #24: Adding cluster requires argocd cli login
Currently, when adding a remote cluster, we call Argo-CD using its cli. It requires the `~/.argocd/config` file to contain the correct data for its cli to connect to Argo-CD server.

Preferably, we will add the cluster using a GitOps approach, and commit whatever Argo-CD requires, to the repo.
Also - on the remote cluster, we might create an app that also manages the SA, RB, CRB resources.

### PR DESCRIPTION
**Pull request by roi-codefresh**
Closes #24

### CODE REVIEW

**Comment by roi-codefresh** [comment]
* got rid of interfacer
* added codegen to ci-cd flow
* added login to argocd after repo bootstrap

File: hack/cmd-docs/main.go [hunk 1]
```diff
@@ -1,22 +1,68 @@
...
+	home      = "/home/user"
+)
+
+var orgHome = os.Getenv("HOME")
 
 func main() {
-	// set HOME env var so that default values involve user's home directory do not depend on the running user.
-	os.Setenv("HOME", "/home/user")
+	log.Printf("org home: %s", orgHome)
+	log.Printf("new home: %s", home)
```
Comment (noam-codefresh) [inline hack/cmd-docs/main.go:25]: Do we need these two lines?

Comment (roi-codefresh) [reply hack/cmd-docs/main.go:25]: hmm, I mean, I could add a `\n`

**Review by noam-codefresh — APPROVED** [review]

**Comment by roi-codefresh** [comment]
* got rid of interfacer
* added codegen to ci-cd flow
* added login to argocd after repo bootstrap
* increased test coverage

**Pull request merged**